src/azure.ts

```diff
@@ -102,8 +120,211 @@ export async function setSubscriptionAsync(context: Context, subscription: strin
         return { succeeded: false, result: null, error: [sr.stderr] };
     }
 }
+export async function getClusterList(context: Context, subscription: string, clusterType: string) : Promise<ActionResult<ClusterInfo[]>> {
+    // log in
+    const login = await setSubscriptionAsync(context, subscription);
+    if (!login.succeeded) {
+        return {
+            actionDescription: 'logging into subscription',
+            result: { succeeded: false, result: [], error: login.error }
+        };
+    }
+
+    // list clusters
+    const clusters = await listClustersAsync(context, clusterType);
+    return {
+        actionDescription: 'listing clusters',
+        result: clusters
+    };
+}
+
+async function listClustersAsync(context: Context, clusterType: string) : Promise<Errorable<ClusterInfo[]>> {
+    let cmd = getListClustersCommand(context, clusterType);
+    const sr = await context.shell.exec(cmd);
+
+    if (sr.code === 0 && !sr.stderr) {
+        const clusters : ClusterInfo[] = JSON.parse(sr.stdout);
+        return { succeeded: true, result: clusters, error: [] };
```

Comment:
i saw there were many places to return the same formatted success or failure data. Can we  provide two util functions for them? such as `wrapAsSuccessResult(result: T): Errorable<T>`, `wrapAsFailureResult(result: T, error: string[]): Errorable<T>`.